cmake/Modules/GrTest.cmake

```diff
@@ -141,3 +141,26 @@ function(GR_ADD_TEST test_name)
     endif(WIN32)
 
 endfunction(GR_ADD_TEST)
+
+########################################################################
+# Add a C++ unit test and setup the environment for a unit test.
+# Takes the same arguments as the ADD_TEST function.
+#
+# test_name -- An identifier for your test, for usage with ctest -R
+# test_source -- Path to the .cc file
+#
+# Before calling set the following variables:
+# GR_TEST_TARGET_DEPS  - built targets for the library path
```

Comment:
why not make this a parameter and make it more "functionally" programmed?

Reply:
(I'd accept a "because it's not the way we deal with this kind of thing around our CMake code, especially in this particular file" as valid explanation. So, this is not a blocker.)